frontend/app_flowy/lib/workspace/presentation/settings/widgets/settings_appearance_view.dart

```diff
@@ -28,6 +30,124 @@ class SettingsAppearanceView extends StatelessWidget {
   }
 }
 
+class ThemeTypeSetting extends StatelessWidget {
+  final ThemeType currentThemeType;
+  const ThemeTypeSetting({required this.currentThemeType, super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      children: [
+        Expanded(
+          child: FlowyText.medium(
+            LocaleKeys.settings_appearance_themeType_label.tr(),
+            overflow: TextOverflow.ellipsis,
+          ),
+        ),
+        AppFlowyPopover(
+          direction: PopoverDirection.bottomWithRightAligned,
+          child: FlowyTextButton(
+            _getThemeName(currentThemeType),
```

Comment:
It's better to add an extension function for `ThemeType` to get the name.

```dart
extension ThemeTypeExtension on ThemeType {
  String get name => ...
}
```

Reply:
So what I have done here is I have created two copies of this method. One is an extension function as you've suggested, that gets a constant string value independent of the translation logic (since we can't access the LocaleKey file from the flowy-infra package). And another variant of the same method that shows the translated string.